File: problems/3002/solution.rb
require 'set'

# @return {Integer}
def maximum_set_size(nums1, nums2)
  s1, s2 = nums1.to_set, nums2.to_set
  half = nums1.length / 2
  [[s1.length, half].min + [s2.length, half].min, (s1 | s2).length].min
end

File: problems/3002/Solution.scala
object Solution {
    def maximumSetSize(nums1: Array[Int], nums2: Array[Int]): Int = {
        val s1 = nums1.toSet
        val s2 = nums2.toSet
        val half = nums1.length / 2
        math.min((s1 union s2).size, math.min(s1.size, half) + math.min(s2.size, half))
    }
}
